File: 0217-contains-duplicate/main.cpp
/*
 * C++ Hash table solution
 * */
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, bool> hashmap;
        for (int i = 0; i < nums.size(); i++) {
            if (hashmap[nums[i]]) {
                return true;
            }
            hashmap[nums[i]] = true;
        }
        return false;
    }
};

File: 0217-contains-duplicate/main.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # hashset
        # time: O(N)
        # space: O(N)
        hashset = set()
        for i in nums:
            if i in hashset:
                return True
            hashset.add(i)
        return False

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # sort in place
        # time: O(nlogn)
        # space: O(1)
        nums.sort()
        for i in range(1,len(nums)):
            if (nums[i-1] == nums[i]):
                return True

        return False
